src/utils/intercom.ts

```diff
@@ -1,19 +1,23 @@
 import crypto from 'crypto'
+import { CookieAttributes } from 'js-cookie'
 import { COOKIES_KEY_INTERCOM } from 'src/logic/cookies/model/cookie'
 import { loadFromCookie, saveCookie } from 'src/logic/cookies/utils'
 import { INTERCOM_ID } from 'src/utils/constants'
 
 let intercomLoaded = false
 
-export const isIntercomLoaded = () => intercomLoaded
+export const isIntercomLoaded = (): boolean => intercomLoaded
 
 const getIntercomUserId = async () => {
-  const cookiesState = await loadFromCookie(COOKIES_KEY_INTERCOM)
+  const cookiesState = await loadFromCookie(COOKIES_KEY_INTERCOM, true)
   if (!cookiesState) {
     const userId = crypto.randomBytes(32).toString('hex')
     const newCookieState = { userId }
-    const expirationDays = 365
-    await saveCookie(COOKIES_KEY_INTERCOM, newCookieState, expirationDays)
+    const cookieConfig: CookieAttributes = {
+      expires: 365,
+      domain: 'gnosis-safe.io',
```

Comment:
This should be probably conditional based on the env?
Or maybe taken from the [config](https://github.com/gnosis/safe-react/blob/dev/src/config/networks/mainnet.ts#L8)?

Reply:
Or just as simple as
```
domain: isProduction ? 'gnosis-safe.io' : undefined
```